imsim/batoid_wcs.py

```diff
@@ -332,6 +338,12 @@ def _focal_to_field(self, fpx, fpy, z_offset=0.0):
         thx, thy : array
             Field angle in radians
         """
+        det_telescope = self.telescope
+        if z_offset != 0.0:
+            det_telescope = det_telescope.withLocallyShiftedOptic(
+                "Detector", [0.0, 0.0, -z_offset]  # batoid convention is opposite of DM
+            )
```

Comment:
Admittedly, this is only 5 lines, one of which is just a closing paren, but it's repeated twice.  Should we make it quick helper function somewhere?  E.g.
```
    def _get_det_telescope(self, z_offset, _telescope=None):
        if _telescope is not None:
            return _telescope
        elif z_offset != 0.0:
            return self.telescope.withLocallyShiftedOptic(
                "Detector", [0.0, 0.0, -z_offset]  # batoid convention is opposite of DM
            )
        else:
            return self.telescope
```